rules/import-style.js

```diff
@@ -127,6 +128,169 @@ const defaultStyles = {
+		return;
+	}
+
+	const isImportDeclaration = node.type === 'ImportDeclaration';
+	const isVariableDeclarator = node.type === 'VariableDeclarator';
+	const isRequireCall = isCallExpression(node.init, {name: 'require'});
+	const isImportExpression = node.init?.type === 'ImportExpression';
+	const isVariableDeclaratorWithRequireCall = isVariableDeclarator && (isRequireCall || isImportExpression);
+	const isValidNode = isImportDeclaration || isVariableDeclaratorWithRequireCall;
+
+	if (!isValidNode) {
+		return;
+	}
+
+	let importedNames = [];
+	if (node.type === 'ImportDeclaration') {
+		importedNames = node.specifiers
+			.filter(specifier => specifier.type === 'ImportSpecifier' || specifier.type === 'ImportDefaultSpecifier')
+			.map(specifier => ({
+				localName: specifier.local.name,
+				importedName: specifier.type === 'ImportDefaultSpecifier' ? 'default' : specifier.imported.name,
+			}));
+	} else if (node.type === 'VariableDeclarator' && node.id.type === 'ObjectPattern') {
+		importedNames = node.id.properties
+			.map(property => {
+				if (property.type === 'RestElement') {
+					return {
+						localName: property.argument.name,
+						importedName: undefined,
+					};
+				}
+
+				return {
+					localName: property.value.name,
+					importedName: property.key.name,
+				};
+			});
+	}
+
+	if (importedNames.length === 0) {
+		return;
+	}
+
+	return function * (fixer) {
+		const scope = sourceCode.getScope(node);
+
+		const namespaceIdentifier = getNamespaceIdentifier(moduleName);
+
+		// Check if any of the named imports match our desired namespace identifier
+		const hasMatchingNamedImport = importedNames.some(
+			({localName}) => localName === namespaceIdentifier,
+		);
+
+		// Only avoid capture if there's no matching named import
+		const uniqueNamespaceIdentifier = hasMatchingNamedImport
+			? namespaceIdentifier
+			: getAvailableVariableName(namespaceIdentifier, [scope]);
+
+		// For VariableDeclarator, we need to handle the parent VariableDeclaration
+		const hasSemicolon = sourceCode.getText(
+			node.type === 'VariableDeclarator' ? node.parent : node,
+		).endsWith(';');
+
+		yield fixer.replaceTextRange(
+			node.type === 'VariableDeclarator'
+				? [node.parent.range[0], node.parent.range[1]]
+				: [node.range[0], node.range[1]],
+			node.type === 'ImportDeclaration'
+				? `import * as ${uniqueNamespaceIdentifier} from ${sourceCode.getText(node.source)}${hasSemicolon ? ';' : ''}`
+				: `const ${uniqueNamespaceIdentifier} = require(${sourceCode.getText(
+					node.type === 'VariableDeclarator' ? node.init.arguments[0] : node.arguments[0],
+				)})${hasSemicolon ? ';' : ''}`,
+		);
+
+		if (hasMatchingNamedImport) {
+			return;
+		}
+
+		for (const {localName, importedName} of importedNames) {
+			// Skip rest patterns since they should be kept as is
+			if (importedName === undefined) {
+				continue;
+			}
+
+			const programScope = sourceCode.getScope(sourceCode.ast);
+
+			const getAllReferences = scope => {
```

Comment:
This part should use `sourceCode.getDeclaredVariables()` to get variable from the `ImportDeclaration` and then we have all the references.

Reply:
`consistent-assert` has similar logic, you can take a look. https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2535/files